api/web3server_marshal.go

```diff
@@ -245,11 +257,11 @@ func (obj *getTransactionResult) MarshalJSON() ([]byte, error) {
 		S                string  `json:"s"`
 		V                string  `json:"v"`
 	}{
-		Hash:             "0x" + hex.EncodeToString(obj.receipt.ActionHash[:]),
+		Hash:             "0x" + hex.EncodeToString(txHash),
```

Comment:
follow the specifications https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gettransactionbyhash

Reply:
Thx for the correction: The three fields (i.e. `blockHash`, `blockNumber` and `transactionIndex`) should be `null` if transaction is pending.
Here is example call from eth:
```
$ curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x7443280c0df6c69fb643ffde0534508dcefa688906ff0c2dc45041548b6418df"],"id":0}' https://ethereum.iotex.one/v1/mainnet

{
   "id" : 0,
   "jsonrpc" : "2.0",
   "result" : {
      "blockHash" : null,
      "blockNumber" : null,
      "chainId" : "0x1",
      "from" : "0x2e05a304d3040f1399c8c20d2a9f659ae7521058",
      "gas" : "0x1fbd0",
      "gasPrice" : "0x304fed86c",
      "hash" : "0x7443280c0df6c69fb643ffde0534508dcefa688906ff0c2dc45041548b6418df",
      "input" : "0xa5e387511b44474377bfe2d5a56bf69a2e279de9b29c54b7ec78aa0353be740c81a6c425",
      "nonce" : "0x71d7d",
      "r" : "0xae22f44be8e3cf52504e3cba3b8852774021fe4588535df796a0170707a61732",
      "s" : "0x70be719fc2b35eabc302695ca7d0e020fdcdf5104ec14eb9a13913fec2653cff",
      "to" : "0x5be1de8021cc883456fd11dc5cd3806dbc48d304",
      "transactionIndex" : null,
      "type" : "0x0",
      "v" : "0x26",
      "value" : "0x0"
   }
}
```